### ISSUE TITLE
Exception in plugin

### ISSUE THREAD

**Issue by jcassee** [eltonsandre/intellij-spring-boot-helper#2]
Just saw this exception in the IntelliJ logs. Hope it helps.

Spring Boot Helper 2022.2.1
IntelliJ Build #IU-222.3345.118, built on July 26, 2022
Runtime version: 17.0.3+7-b469.32 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
OS: Arch Linux

```
Cannot create extension without pico container (class=dev.eltonsandre.intellij.spring.boot.SpringBootApplicationPresentationProvider, constructors=[public dev.eltonsandre.intellij.spring.boot.SpringBootApplicationPresentationProvider()]), please remove extra constructor parameters

com.intellij.diagnostic.PluginException: Cannot create class dev.eltonsandre.intellij.spring.boot.SpringBootApplicationPresentationProvider (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Spring Boot Helper, id=dev.eltonsandre.intellij.spring.assistant.plugin, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/IntelliJIdea2022.2/intellij-spring-boot, version=2022.2.1, package=null, isBundled=false), packagePrefix=null, instanceId=402, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:883)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.instantiateClass(XmlExtensionAdapter.kt:79)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.kt:90)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.doCreateInstance(XmlExtensionAdapter.kt:50)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.kt:31)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:475)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:423)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:266)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensions(ExtensionPointName.java:35)
	at com.intellij.openapi.roots.impl.libraries.LibraryDetectionManagerImpl.computeKinds(LibraryDetectionManagerImpl.java:65)
	at com.intellij.openapi.roots.impl.libraries.LibraryDetectionManagerImpl.getOrComputeKinds(LibraryDetectionManagerImpl.java:56)
	at com.intellij.openapi.roots.impl.libraries.LibraryDetectionManagerImpl.processProperties(LibraryDetectionManagerImpl.java:29)
	at com.intellij.openapi.roots.ui.configuration.libraries.impl.LibraryPresentationManagerImpl.getCustomIcons(LibraryPresentationManagerImpl.java:98)
	at com.intellij.openapi.roots.ui.configuration.libraries.impl.LibraryPresentationManagerImpl.getCustomIcon(LibraryPresentationManagerImpl.java:86)
	at com.intellij.openapi.roots.ui.configuration.libraries.impl.LibraryPresentationManagerImpl.getNamedLibraryIcon(LibraryPresentationManagerImpl.java:75)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.LibraryConfigurable.getIcon(LibraryConfigurable.java:157)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectStructureElementRenderer.customizeCellRenderer(ProjectStructureElementRenderer.java:51)
	at com.intellij.ui.ColoredTreeCellRenderer.rendererComponentInner(ColoredTreeCellRenderer.java:125)
	at com.intellij.ui.ColoredTreeCellRenderer.getTreeCellRendererComponent(ColoredTreeCellRenderer.java:57)
	at com.intellij.ui.tree.ui.DefaultTreeUI.getRenderer(DefaultTreeUI.java:163)
	at com.intellij.ui.tree.ui.DefaultTreeUI$2.getNodeDimensions(DefaultTreeUI.java:506)
	at java.desktop/javax.swing.tree.AbstractLayoutCache.getNodeDimensions(AbstractLayoutCache.java:497)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.updatePreferredSize(VariableHeightLayoutCache.java:1344)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1465)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1272)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.rebuild(VariableHeightLayoutCache.java:728)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.treeStructureChanged(VariableHeightLayoutCache.java:629)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeStructureChanged(BasicTreeUI.java:4407)
	at java.desktop/javax.swing.tree.DefaultTreeModel.fireTreeStructureChanged(DefaultTreeModel.java:616)
	at java.desktop/javax.swing.tree.DefaultTreeModel.reload(DefaultTreeModel.java:320)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.BaseLibrariesConfigurable.createLibrariesNode(BaseLibrariesConfigurable.java:137)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.BaseLibrariesConfigurable.loadTree(BaseLibrariesConfigurable.java:103)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.BaseStructureConfigurable.loadTreeNodes(BaseStructureConfigurable.java:226)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.BaseStructureConfigurable.reset(BaseStructureConfigurable.java:216)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.BaseLibrariesConfigurable.reset(BaseLibrariesConfigurable.java:97)
	at com.intellij.openapi.roots.ui.configuration.ProjectStructureConfigurable.lambda$reset$1(ProjectStructureConfigurable.java:347)
	at com.intellij.util.io.storage.HeavyProcessLatch.performOperation(HeavyProcessLatch.java:76)
	at com.intellij.openapi.roots.ui.configuration.ProjectStructureConfigurable.reset(ProjectStructureConfigurable.java:333)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.reset(ConfigurableCardPanel.java:193)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.lambda$createConfigurableComponent$4(ConfigurableCardPanel.java:125)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:941)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.createConfigurableComponent(ConfigurableCardPanel.java:112)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.create(ConfigurableCardPanel.java:60)
	at com.intellij.openapi.options.newEditor.ConfigurableEditor$1.create(ConfigurableEditor.java:57)
	at com.intellij.openapi.options.newEditor.ConfigurableEditor$1.create(ConfigurableEditor.java:54)
	at com.intellij.ui.CardLayoutPanel.createValue(CardLayoutPanel.java:75)
	at com.intellij.ui.CardLayoutPanel.select(CardLayoutPanel.java:103)
	at com.intellij.ui.CardLayoutPanel.select(CardLayoutPanel.java:88)
	at com.intellij.openapi.options.newEditor.ConfigurableEditor.init(ConfigurableEditor.java:108)
	at com.intellij.openapi.options.newEditor.SingleSettingEditor.<init>(SingleSettingEditor.java:29)
	at com.intellij.openapi.options.newEditor.SettingsDialog.<init>(SettingsDialog.java:47)
	at com.intellij.openapi.options.newEditor.SettingsDialogFactory.create(SettingsDialogFactory.java:23)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.editConfigurable(ShowSettingsUtilImpl.java:264)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.editConfigurable(ShowSettingsUtilImpl.java:253)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.editConfigurable(ShowSettingsUtilImpl.java:211)
	at com.intellij.openapi.roots.ui.configuration.ModulesConfigurator.showDialog(ModulesConfigurator.java:545)
	at com.intellij.openapi.roots.ui.configuration.IdeaProjectSettingsService.openModuleSettings(IdeaProjectSettingsService.java:54)
	at com.android.tools.idea.gradle.structure.AndroidProjectSettingsServiceImpl.openModuleSettings(AndroidProjectSettingsServiceImpl.java:117)
	at com.intellij.ide.projectView.impl.nodes.PsiDirectoryNode.navigate(PsiDirectoryNode.java:292)
	at com.intellij.util.OpenSourceUtil.navigate(OpenSourceUtil.java:90)
	at com.intellij.util.OpenSourceUtil.navigate(OpenSourceUtil.java:59)
	at com.intellij.util.OpenSourceUtil.navigate(OpenSourceUtil.java:52)
	at com.intellij.util.OpenSourceUtil.openSourcesFrom(OpenSourceUtil.java:20)
	at com.intellij.util.EditSourceOnDoubleClickHandler$TreeMouseListener.processDoubleClick(EditSourceOnDoubleClickHandler.java:172)
	at com.intellij.util.EditSourceOnDoubleClickHandler$TreeMouseListener.onDoubleClick(EditSourceOnDoubleClickHandler.java:166)
	at com.intellij.ui.DoubleClickListener.onClick(DoubleClickListener.java:30)
	at com.intellij.ui.ClickListener$1.mouseReleased(ClickListener.java:59)
	at com.intellij.openapi.wm.impl.IdeGlassPaneImpl.fireMouseEvent(IdeGlassPaneImpl.java:438)
	at com.intellij.openapi.wm.impl.IdeGlassPaneImpl.dispatchMouseReleased(IdeGlassPaneImpl.java:244)
	at com.intellij.openapi.wm.impl.IdeGlassPaneImpl.dispatchForDnDAware(IdeGlassPaneImpl.java:219)
	at com.intellij.openapi.wm.impl.IdeGlassPaneImpl.dispatchMouseEvent(IdeGlassPaneImpl.java:138)
	at com.intellij.openapi.wm.impl.IdeGlassPaneImpl.dispatch(IdeGlassPaneImpl.java:90)
	at com.intellij.ide.IdeEventQueue.dispatchByCustomDispatchers(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:721)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:791)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:113)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:881)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.IllegalArgumentException: Kind spring-boot is not unique
	at com.intellij.openapi.roots.libraries.LibraryKind.<init>(LibraryKind.java:37)
	at com.intellij.openapi.roots.libraries.LibraryKind.create(LibraryKind.java:56)
	at dev.eltonsandre.intellij.spring.boot.SpringBootApplicationPresentationProvider.<init>(Unknown Source)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:844)
	... 88 more
```

**Comment by eltonsandre** [comment]
Hi @jcassee,

thanks for reporting!
I will investigate and apply a fix.
thanks you!

**Comment by eltonsandre** [comment]
Hi @jcassee ,

Fixed in plugin version 2022.2.2

**Comment by jcassee** [comment]
Thanks, Elton!
